api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -1848,6 +1852,55 @@ public void savePatient_shouldUpdateAnExistingPatient() throws Exception {
 		assertTrue("The gender should be new", patient2.getGender().equals("F"));
 	}
 	
+	@Test
+	public void savePatient_shouldFailSaveIfPatientsWithSameIdentifierAreSaved() {
+	    isSavePatientThreadsafe = new AtomicBoolean(false);
+		UserContext ctx = Context.getUserContext();
+		
+		Patient patient1 = new Patient();
+		patient1.setGender("M");
+		patient1.addName(new PersonName("Test1", "Sam", "Test"));
+		
+		Patient patient2 = new Patient();
+		patient2.setGender("M");
+		patient2.addName(new PersonName("Test2", "Samuel", "Test"));
+		
+		Runnable r1 = () -> savePatientOnThread(ctx, patient1);
+		       
+		Runnable r2 = () -> savePatientOnThread(ctx, patient2);
+				
+		Thread t1 = new Thread(r1);
+		Thread t2 = new Thread(r2);
+		
+		t1.start();
+		t2.start();
+		
+		try {
+			t1.join();
+			//t2.join();
```

Comment:
Did you take a look at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotcommentoutcode